contrib_bots/lib/poll.py

```diff
@@ -0,0 +1,105 @@
+
+
+class PollHandler(object):
+    '''
+    This plugin facilitates polling.
+    '''
+
+    def __init__(self):
+        # polls is a dict of dict. First dict corresponds to poll title,
+        # second refers to answers in poll
+        self.polls = dict()
+
+        # streams is a convenience object that keeps track of the latest
+        # question for each stream.
+        self.streams = dict()
+
+    def usage(self):
+        return '''
+        This plugin facilitates polling.
+
+        Commands:
+        @question poll title: option 1, option 2, option 3
+        A valid poll must use a colon (:) to devide question from answers.
+        Answers must be seperate by coma (,).
+
+        @answer option 1
+        Question will be infered by scanning channel for last poll with
+        valid answer.
+
+        @result poll title
+        Returns a formatted version of the poll.
+
+        >> @question what's the best ice cream flavour: vanilla, chocolate
+        << Question successfully created! Answer away
+        >> @answer vanilla
+        << vanilla has 1 vote
+        >> @result what's the best ice cream flavour
+        << what's the best ice cream flavour:
+        << vanilla: 1
+        << chocolate: 0
+        '''
+
+    # Returns True if message is respondable
+    def triage_message(self, message):
+        original_content = message['content'].lower()
+
+        if message['display_recipient'] == 'poll':
+            return False
+
+        if original_content.startswith('@question'):
```

Comment:
I've suggested the `@poll init` since we're using it in a different realm for an `rsvp` bot (setting up meetings) and I'm just used to it, you can definitely use `start` or anything that works :)